phpBB/phpbb/db/migration/tool/permission.php

```diff
@@ -425,13 +425,27 @@ public function permission_set($name, $auth_option, $type = 'role', $has_permiss
 				$role_id = (int) $this->db->sql_fetchfield('auth_role_id');
 				if ($role_id)
 				{
-					$sql = 'SELECT role_name
+					$sql = 'SELECT role_name, role_type
 						FROM ' . ACL_ROLES_TABLE . '
 						WHERE role_id = ' . $role_id;
 					$this->db->sql_query($sql);
-					$role_name = $this->db->sql_fetchfield('role_name');
-
-					return $this->permission_set($role_name, $auth_option, 'role', $has_permission);
+					$role_data = $this->db->sql_fetchrow();
+					$role_name = $role_data['role_name'];
+					$role_type = $role_data['role_type'];
+
+					// Filter new auth options to match the role type: a_ | f_| m_ | u_
+					// Set new auth options to the role only if options matching the role type were found
+					$auth_option = array_filter($auth_option,
+						function ($option) use ($role_type)
+						{
+							return strpos($auth_option, $auth_type) === 0;
```

Comment:
But you're passing the $auth_option array values it in to the closure as `$option` which is what is being tested in strpos...no?